### PR TITLE
Hurl integration tests plumbing & CI setup

### PR DESCRIPTION
**Pull request by rosado**
We'd like to use Hurl scripts as a way to automate and run integration tests on the ldapi service.

This PR contains mostly changes that allow us to do that both locally and in CI. 

## Changes to the application code
- We now check the `CI` environment variable and disable authentication if it's set to `true`. This allows us to run the built container in CI and run test against it.

## Changes to test code
- a new test source path: `test-hurl` for running hurl scripts. Right now it's located at `./datahost-ld-openapi/test-hurl` but I'm thinking of nesting different kinds of tests under  `test/main`, `test/hurl` etc.
- a utility namespace for launching hurl (via `clojure.java.shell/sh`) and dealing with variables. The safest approach is to launch generate new, unique variables for each script (as everything is scoped to a series anyway).
- only a few regression tests were added (scripts I used to verify fixes for bugs). More are in the works (but will come as different PR).
- documentation for the hurl setup in `datahost-ld-openapi/doc/integration-testing.md`

## Changes to CI setup
- more details on that in the [comment][issue]. tldr: added new jobs for running hurl tests.
- better names for existing jobs
- the hurl tests are run via a babashka script (`datahost-ld-openapi/bin/hurl-runner.bb`)

[issue]:https://github.com/Swirrl/datahost-prototypes/issues/276#issuecomment-1782600987

### CODE REVIEW

File: datahost-ld-openapi/test-hurl/tpximpact/datahost/ldapi/integration_test.clj [hunk 1]
```diff
@@ -0,0 +1,27 @@
+(ns tpximpact.datahost.ldapi.integration-test
```
Comment (RickMoynihan) [inline datahost-ld-openapi/test-hurl/tpximpact/datahost/ldapi/integration_test.clj:1]: flagged re test runner comment.

File: datahost-ld-openapi/test/tpximpact/datahost/ldapi/test_util/hurl.clj [hunk 2]
```diff
@@ -0,0 +1,133 @@
+(ns tpximpact.datahost.ldapi.test-util.hurl
```
Comment (RickMoynihan) [inline datahost-ld-openapi/test/tpximpact/datahost/ldapi/test_util/hurl.clj:1]: flagged re test runner comment

File: .circleci/config.yml [hunk 3]
```diff
@@ -99,162 +94,234 @@ jobs:
...
+    # runs ldapi service container + container with hurl
+    parameters:
+      commit_sha:
+        type: string
+        description: SHA of the git commit used as docker image tag.
+      project_dir:
+        description: Project directory
+        type: string
+    docker:
+      - image: rosado/ci-clojure-hurl:1.11.1.1
```
[the remark this answers is not in the document]
Comment (rosado) [reply .circleci/config.yml:163]: moved the dockerfile to dev-containers repo

File: .circleci/scripts/wait-for-service [hunk 4]
```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (rosado) [reply .circleci/scripts/wait-for-service]: thanks, I'm stealing this.

File: datahost-ld-openapi/bb.edn [hunk 5]
```diff
@@ -0,0 +1 @@
+{:paths ["test"]}
```
[the remark this answers is not in the document]
Comment (rosado) [reply datahost-ld-openapi/bb.edn:1]: As discussed, the code for running all scripts in a directory is shared between the `test-hurl` and `test` aliases.

**Pull request merged**